org.jacoco.ant.test/src/org/jacoco/ant/InstrumentTaskTest.xml

```diff
@@ -123,5 +123,24 @@
 		</java>
 		<au:assertFileExists file="${temp.dir}/test.exec" />
 	</target>
+
+	<target name="testInstrumentWithRuntimeStartupFailure">
+		<jacoco:instrument destdir="${temp.dir}">
+			<fileset dir="${org.jacoco.ant.instrumentTaskTest.classes.dir}" includes="**/*.class"/>
+		</jacoco:instrument>
+		<au:assertLogContains text="Instrumented 15 classes to ${temp.dir}"/>
+		<au:assertFileExists file="${temp.dir}/org/jacoco/ant/InstrumentTaskTest.class" />
+
+		<java classname="org.jacoco.ant.TestTarget" failonerror="false" fork="true">
+			<sysproperty key="jacoco-agent.output" value="tcpserver"/>
+			<sysproperty key="jacoco-agent.port" value="foo"/>
+			<classpath>
+				<pathelement path="${org.jacoco.ant.instrumentTaskTest.agent.file}"/>
+				<pathelement path="${temp.dir}"/>
+			</classpath>
+		</java>
+		<au:assertLogContains text="Caused by: java.lang.RuntimeException: Failed to initialize JaCoCo."/>
```

Comment:
@marchof indeed now JVMs do not crash! :thumbsup: and test succeeds for Java 11+ class files, because access is performed via bootstrap method for condy and so exception indeed looks as

```
Exception in thread "main" java.lang.BootstrapMethodError: bootstrap method initialization exception
    at java.base/java.lang.invoke.BootstrapMethodInvoker.invoke(BootstrapMethodInvoker.java:194)
    at java.base/java.lang.invoke.ConstantBootstraps.makeConstant(ConstantBootstraps.java:67)
    at java.base/java.lang.invoke.MethodHandleNatives.linkDynamicConstantImpl(MethodHandleNatives.java:314)
    at java.base/java.lang.invoke.MethodHandleNatives.linkDynamicConstant(MethodHandleNatives.java:306)
    at org.jacoco.ant.TestTarget.main(TestTarget.java)
Caused by: java.lang.RuntimeException: Failed to initialize JaCoCo.
    at org.jacoco.agent.rt.internal_9f79065.Offline.getRuntimeData(Offline.java:40)
    at org.jacoco.agent.rt.internal_9f79065.Offline.getProbes(Offline.java:59)
    at org.jacoco.ant.TestTarget.$jacocoInit(TestTarget.java)
    at java.base/java.lang.invoke.BootstrapMethodInvoker.invoke(BootstrapMethodInvoker.java:204)
    at java.base/java.lang.invoke.BootstrapMethodInvoker.invoke(BootstrapMethodInvoker.java:90)
    ... 4 more
Caused by: java.lang.NumberFormatException: For input string: "foo"
    at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
    at java.base/java.lang.Integer.parseInt(Integer.java:652)
    at java.base/java.lang.Integer.parseInt(Integer.java:770)
    at org.jacoco.agent.rt.internal_9f79065.core.runtime.AgentOptions.getOption(AgentOptions.java:583)
    at org.jacoco.agent.rt.internal_9f79065.core.runtime.AgentOptions.getPort(AgentOptions.java:452)
    at org.jacoco.agent.rt.internal_9f79065.output.TcpServerOutput.createServerSocket(TcpServerOutput.java:106)
    at org.jacoco.agent.rt.internal_9f79065.output.TcpServerOutput.startup(TcpServerOutput.java:53)
    at org.jacoco.agent.rt.internal_9f79065.Agent.startup(Agent.java:127)
    at org.jacoco.agent.rt.internal_9f79065.Agent.getInstance(Agent.java:53)
    at org.jacoco.agent.rt.internal_9f79065.Offline.getRuntimeData(Offline.java:38)
    ... 8 more
```

However test fails for class files of lower versions, because access is direct and so exception looks a bit differently

```
Exception in thread "main" java.lang.RuntimeException: Failed to initialize JaCoCo.
    at org.jacoco.agent.rt.internal_9f79065.Offline.getRuntimeData(Offline.java:40)
    at org.jacoco.agent.rt.internal_9f79065.Offline.getProbes(Offline.java:59)
    at org.jacoco.ant.TestTarget.$jacocoInit(TestTarget.java)
    at org.jacoco.ant.TestTarget.main(TestTarget.java)
Caused by: java.lang.NumberFormatException: For input string: "foo"
    at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
    at java.base/java.lang.Integer.parseInt(Integer.java:652)
    at java.base/java.lang.Integer.parseInt(Integer.java:770)
    at org.jacoco.agent.rt.internal_9f79065.core.runtime.AgentOptions.getOption(AgentOptions.java:583)
    at org.jacoco.agent.rt.internal_9f79065.core.runtime.AgentOptions.getPort(AgentOptions.java:452)
    at org.jacoco.agent.rt.internal_9f79065.output.TcpServerOutput.createServerSocket(TcpServerOutput.java:106)
    at org.jacoco.agent.rt.internal_9f79065.output.TcpServerOutput.startup(TcpServerOutput.java:53)
    at org.jacoco.agent.rt.internal_9f79065.Agent.startup(Agent.java:127)
    at org.jacoco.agent.rt.internal_9f79065.Agent.getInstance(Agent.java:53)
    at org.jacoco.agent.rt.internal_9f79065.Offline.getRuntimeData(Offline.java:38)
    ... 3 more
```

:wink:

Maybe we can simply remove first prefix `Caused by:` from the test

```suggestion
		<au:assertLogContains text="java.lang.RuntimeException: Failed to initialize JaCoCo."/>
```

?

Reply:
@Godin fixed